bottlerocket/samples/Makefile.toml

```diff
@@ -0,0 +1,242 @@
+ECS_RESOURCE_AGENT_IMAGE_URI = { value = "${AGENT_IMAGE_REGISTRY}/ecs-resource-agent:v${AGENT_IMAGE_VERSION}", condition = { env_not_set = ["ECS_RESOURCE_AGENT_IMAGE_URI"] } }
+EC2_RESOURCE_AGENT_IMAGE_URI = { value = "${AGENT_IMAGE_REGISTRY}/ec2-resource-agent:v${AGENT_IMAGE_VERSION}", condition = { env_not_set = ["EC2_RESOURCE_AGENT_IMAGE_URI"] } }
+ECS_WORKLOAD_AGENT_IMAGE_URI = { value = "${AGENT_IMAGE_REGISTRY}/ecs-workload-agent:v${AGENT_IMAGE_VERSION}", condition = { env_not_set = ["ECS_WORKLOAD_AGENT_IMAGE_URI"] } }
+K8S_WORKLOAD_AGENT_IMAGE_URI = { value = "${AGENT_IMAGE_REGISTRY}/k8s-workload-agent:v${AGENT_IMAGE_VERSION}", condition = { env_not_set = ["K8S_WORKLOAD_AGENT_IMAGE_URI"] } }
+SONOBUOY_TEST_AGENT_IMAGE_URI = { value = "${AGENT_IMAGE_REGISTRY}/sonobuoy-test-agent:v${AGENT_IMAGE_VERSION}", condition = { env_not_set = ["SONOBUOY_TEST_AGENT_IMAGE_URI"] } }
+EKS_RESOURCE_AGENT_IMAGE_URI = { value = "${AGENT_IMAGE_REGISTRY}/eks-resource-agent:v${AGENT_IMAGE_VERSION}", condition = { env_not_set = ["EKS_RESOURCE_AGENT_IMAGE_URI"] } }
+VSPHERE_K8S_CLUSTER_RESOURCE_AGENT_IMAGE_URI = { value = "${AGENT_IMAGE_REGISTRY}/vsphere-k8s-cluster-resource-agent:v${AGENT_IMAGE_VERSION}", condition = { env_not_set = ["VSPHERE_K8S_CLUSTER_RESOURCE_AGENT_IMAGE_URI"] } }
+VSPHERE_VM_RESOURCE_AGENT_IMAGE_URI = { value = "${AGENT_IMAGE_REGISTRY}/vsphere-vm-resource-agent:v${AGENT_IMAGE_VERSION}", condition = { env_not_set = ["VSPHERE_VM_RESOURCE_AGENT_IMAGE_URI"] } }
+CONTROLLER_IMAGE_URI = { value = "${AGENT_IMAGE_REGISTRY}/controller:v${AGENT_IMAGE_VERSION}", condition = { env_not_set = ["CONTROLLER_IMAGE_URI"] } }
+
+[tasks.metadata-url.env]
+METADATA_URL = { script=["echo ${METADATA_BASE_URL}/${VARIANT}/${ARCHITECTURE}"], condition = { env_not_set = ["METADATA_URL"] } }
+
+[tasks.cluster-name.env]
+CLUSTER_NAME = { script=["echo $(echo ${ARCHITECTURE} | tr '_' '-')-$(echo ${VARIANT} | tr -d '.')"], condition = { env_not_set = ["CLUSTER_NAME"] } }
+
+[tasks.ova-name.env]
+OVA_NAME = { script=["echo bottlerocket-${VARIANT}-x86_64-${STARTING_VERSION}.ova"], condition = { env_not_set = ["OVA_NAME"] } }
+
+[tasks.bottlerocket-ami-id.env]
+BOTTLEROCKET_AMI_ID = { script=['''
+if [ $(echo ${TEST_TYPE} | tr "-" " " | awk '{print $2}') = "migration" ]; then
+    AMI_VERSION=$(echo ${STARTING_VERSION} | tr -d "v")
+else
+    AMI_VERSION="latest"
+fi
+aws ssm get-parameter \
+    --region ${AWS_REGION} \
+    --name "/aws/service/bottlerocket/${VARIANT}/$(echo ${ARCHITECTURE} | sed -e 's/aarch64/arm64/g')/${AMI_VERSION}/image_id" \
+    --query Parameter.Value --output text 
+'''], condition = { env_not_set = ["BOTTLEROCKET_AMI_ID"], env_contains = { "VARIANT" = "aws" } } }
+
+[tasks.gpu.env]
+GPU = { script=['''
+if [ $(echo ${VARIANT} | tr "-" " " | awk '{print $NF}') = "nvidia" ]; then
```

Comment:
:(

I get it though.

Reply:
I suppose this can be made to look cleaner by using a regex string compare (aka "contains nvidia") but the logic will be the same.

Reply:
Let's use the regex just for clarity.